idaes/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/hetero_reactions.py

```diff
@@ -0,0 +1,517 @@
+Mapping of the range of operational conditions for cu-, Fe-, and Ni-based
+oxygen carriers in chemical-looping combustion,
+Chem. Eng. Sci. 62 (2007) 533–549.
+
+"""
+
+# Import Pyomo libraries
+from pyomo.environ import (Constraint,
+                           exp,
+                           Param,
+                           PositiveReals,
+                           Reals,
+                           Set,
+                           value,
+                           Var)
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+from pyomo.opt import SolverFactory
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+
+
+# Import IDAES cores
+from idaes.core import (declare_process_block_class,
+                        MaterialFlowBasis,
+                        ReactionParameterBlock,
+                        ReactionBlockDataBase,
+                        ReactionBlockBase,
+                        Component,
+                        VaporPhase,
+                        SolidPhase)
+from idaes.core.util.misc import add_object_reference
+from idaes.core.util.initialization import (fix_state_vars,
+                                            revert_state_vars,
+                                            solve_indexed_blocks)
+from idaes.core.util.model_statistics import (
+    number_unfixed_variables_in_activated_equalities)
+from idaes.core.util.config import (is_state_block,
+                                    is_physical_parameter_block,
+                                    is_reaction_parameter_block)
+import idaes.logger as idaeslog
+
+# Some more information about this module
+__author__ = "Chinedu Okoli"
+
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+@declare_process_block_class("HeteroReactionParameterBlock")
+class ReactionParameterData(ReactionParameterBlock):
+    """
+    Property Parameter Block Class
+
+    Contains parameters and indexing sets associated with properties for
+    superheated steam.
+
+    """
+    # Create Class ConfigBlock
+    CONFIG = ConfigBlock()
+    CONFIG.declare("gas_property_package", ConfigValue(
+            description="Reference to associated PropertyPackageParameter "
+                        "object for the gas phase.",
+            domain=is_physical_parameter_block))
+    CONFIG.declare("solid_property_package", ConfigValue(
+            description="Reference to associated PropertyPackageParameter "
+                        "object for the solid phase.",
+            domain=is_physical_parameter_block))
+    CONFIG.declare("default_arguments", ConfigBlock(
+            description="Default arguments to use with Property Package",
+            implicit=True))
+
+    def build(self):
+        '''
+        Callable method for Block construction.
+        '''
+        super(ReactionParameterBlock, self).build()
+
+        self._reaction_block_class = ReactionBlock
+
+        # Create Phase objects
+        self.Vap = VaporPhase()
```

Comment:
1. Reaction blocks should generally not build their own phase and component lists - I think it would be better to point to the original sets in the thermo property packages.
2. `Phase` and `Component` objects were not intended for use in reaction packages. Whilst there is no issue at the moment, we might want to be careful in case something arises in the future.